File: backend/backend-java/src/main/java/com/cayena/backend/dtos/responses/ProductResponse.java
package com.cayena.backend.dtos.responses;

import com.cayena.backend.entities.Product;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductResponse {

    private Long id;
    private String name;
    private Integer quantityInStock;
    private BigDecimal unitPrice;
    private String supplierName;

    public static ProductResponse converter(Product product) {
        ProductResponse response = new ProductResponse();
        response.setId(product.getProductId());
        response.setName(product.getName());
        response.setQuantityInStock(product.getQuantityInStock());
        response.setUnitPrice(product.getUnitPrice());
        response.setSupplierName(product.getSupplier().getName());
        return  response;
    }
}

File: backend/backend-kotlin/src/main/kotlin/com/cayena/backendkotlin/dtos/responses/ProductResponse.kt
package com.cayena.backendkotlin.dtos.responses

import com.cayena.backendkotlin.dominio.Product
import com.cayena.backendkotlin.dominio.Supplier
import java.math.BigDecimal
import java.time.LocalDateTime

class ProductResponse(
    val name: String,
    val quantityInStock: Int,
    val unitPrice: BigDecimal,
    val supplierName: String
){
    fun toEntity(): Product = Product(
        name = this.name,
        quantityInStock = this.quantityInStock,
        unitPrice = this.unitPrice,
        supplier = Supplier(
            supplierId = 1,
            nameSupplier = supplierName,
            dateOfCreation = LocalDateTime.now(),
            dateOfTheLastUpdate = LocalDateTime.now()
        )
    )
}

